evm_arithmetization/src/cpu/columns/general.rs

```diff
@@ -163,3 +183,18 @@ pub(crate) struct CpuStackView<T: Copy> {
 /// Number of columns shared by all the views of `CpuGeneralColumnsView`.
 /// `u8` is guaranteed to have a `size_of` of 1.
 pub(crate) const NUM_SHARED_COLUMNS: usize = size_of::<CpuGeneralColumnsView<u8>>();
+
+/// The number of columns in the largest field of [`CpuGeneralColumnsView`].
+/// We assert that this is the same as [`NUM_SHARED_COLUMNS`], but we define
+/// it in order to determine the number of padding columns to add to each field
+/// without creating a cycle for rustc.
+const NUM_UNION_COLUMNS: usize = size_of::<CpuLogicView<u8>>();
+const_assert!(NUM_UNION_COLUMNS == NUM_SHARED_COLUMNS);
```

Comment:
I don't think this new constant is useful. We can eventually clarify the documentation of `NUM_SHARED_COLUMNS`, define it from `CpuLogicView` if you prefer, and rely on that one directly instead (as they are both essentially the same).